### PR TITLE
Fix #8624: Unable to delete the key when test completed

### LINKED ISSUE #8624: Unable to delete the key when test completed [AzConfig]
After test completion, below key and values are still available in the configuration explorer which should be deleted once test execution completes. 

```
Key= ~`!@#$^&()_+=[]{}|;"'<>./--1114869967
Value= value-~`!@#$^&()_+=[]{}|;"'<>./-
```

### PR DESCRIPTION
**Pull request by AlexanderSher**
No product change is required, tests are updated to ensure that settings created in test are deleted.

### CODE REVIEW

**Comment by pakrym** [comment]
We usually don't check the response status in tests, why is it worth doing here?

**Comment by AlexanderSher** [comment]
Because 204 is a success code, and we want to be sure that entry is in fact deleted (status code 200).

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs [hunk 1]
```diff
@@ -146,7 +146,7 @@ public async Task DeleteSetting()
             }
             finally
             {
-                await service.DeleteConfigurationSettingAsync(testSetting.Key, testSetting.Label);
+                AssertStatus200(await service.DeleteConfigurationSettingAsync(testSetting.Key, testSetting.Label));
```
Comment (pakrym) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs:149]: Why doesn't this fail? We are deleting the secret at https://github.com/Azure/azure-sdk-for-net/pull/11205/files#diff-3aa297bb699ae547859d1b6af28fb9dbR137

Comment (AlexanderSher) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs:149]: `testSettingDiff` has no label.

**Review by pakrym — APPROVED** [review]

**Pull request merged**